repository/auth_repo.go

```diff
@@ -56,3 +71,50 @@ func (a AuthRepo) Get(ctx context.Context, key domain.AuthAPIKey) (string, bool)
 
 	return string(environment), true
 }
+
+// GetKeysForEnvironment gets all the apikey keys associated with environment id
+func (a AuthRepo) GetKeysForEnvironment(ctx context.Context, envID string) ([]string, bool) {
+
+	var apiKeys []string
+
+	key := NewAPIConfigsKey(envID)
+	if err := a.cache.Get(ctx, string(key), &apiKeys); err != nil {
+		return apiKeys, false
+	}
+
+	return apiKeys, true
+}
+
+// RemoveAllKeysForEnvironment all api keys for given environment
+func (a AuthRepo) RemoveAllKeysForEnvironment(ctx context.Context, envID string) error {
+
+	apiKeys, ok := a.GetKeysForEnvironment(ctx, envID)
+	if !ok {
+		return fmt.Errorf("unable to get apiKeys for environment: %v", envID)
+	}
+
+	// append the entry for the list of keys assocaited with environments
+	// we do that to delete them all in the next step.
+	key := NewAPIConfigsKey(envID)
+	apiKeys = append(apiKeys, string(key))
+
+	//remove entries for all keys associated with environments
+	return a.Remove(ctx, apiKeys)
+}
+
+// Remove removes from cache all provided keys
+func (a AuthRepo) Remove(ctx context.Context, keys []string) error {
+
+	for _, k := range keys {
+		if err := a.cache.Delete(ctx, k); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+type APIConfigsKey string
```

Comment:
Should we move this and the constructor below to the domain package beside the other Key types?